disperser/batcher/batcher.go

```diff
@@ -493,9 +489,12 @@ func (b *Batcher) HandleSingleBatch(ctx context.Context) error {
 		return fmt.Errorf("HandleSingleBatch: error sending confirmBatch transaction: %w", err)
 	} else {
 		for _, metadata := range batch.BlobMetadata {
-			err = b.EncodingStreamer.MarkBlobPendingConfirmation(metadata)
+			err = b.Queue.MarkBlobConfirming(ctx, metadata.GetBlobKey())
```

Comment:
I'm confused about the error handling. Here we check if marking the blobs in blobstore is successful before removing the encoded blobs. In `handleFailure` (e.g. L488), we remove the encoded blobs first and then update the blobstore.  Have we thought through the different states that the system can be in depending on these different failure scenarios?

Reply:
This is a good point. The encoded blob should probably be removed only if the update is successful.
The difference is whether the blob gets encoded again or not, which isn't super critical but has important performance implications.

Reply:
I think that's a good idea. 
I made `transitionBlobToConfirming` but kept the transitions to `Failed` and `Confirmed` as they were because those transitions only update blob store.